wfe2/wfe_test.go

```diff
@@ -1746,8 +1754,52 @@ func TestAccount(t *testing.T) {
 	}`)
 }
 
+type mockSAWithCert struct {
+	core.StorageGetter
+	cert   *x509.Certificate
+	status core.OCSPStatus
+}
+
+func newMockSAWithCert(t *testing.T, sa core.StorageGetter, valid bool) *mockSAWithCert {
```

Comment:
Nit: I like to avoid bool params when possible, because it's hard to intuit at the call site what they're for. Here we already have a handy type: `core.OCSPStatus`. We could replace `valid` with `status`. Alternately, we could have a specialization `newMockSAWithRevokedCert` that sets the revoked status.

Reply:
I considered but dismissed a separate mock for revoked cert, because I felt like the boilerplate-to-benefit ratio was too high. Taking a status works, though.